src/Suggest/CandidateGenerator/DirectGenerator.php

```diff
@@ -132,4 +132,20 @@ public function setPostFilter(string $post)
     {
         return $this->setParam('post_filter', $post);
     }
+
+    /**
+     * Convert to array so the generator can be included in a direct_generator list.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $data = $this->getParams();
+
+        if (!empty($this->_rawParams)) {
+            $data = \array_merge($data, $this->_rawParams);
+        }
+
+        return $this->_convertArrayable($data);
```

Comment:
Implementing NameableInterface here should do the trick and avoid customizing the `toArray()`

Reply:
@deguif Thanks for suggesting the changes and align this with the future changes for the lib.

 I do not see how implementing NameableInterface will do the trick here. The handling for that interface is:  
`$arr[$value instanceof NameableInterface ? $value->getName() : $key] = $value->toArray();` So we will still get a map with the "direct_generator" name and params, instead of a list of direct_generator's params.

We are looking to get 
```
{
    "suggest": {
        "text": "obel prize",
        "simple_phrase": {
            "phrase": {
                "field": "title.trigram",
                "size": 1,
                "direct_generator": [
                    {
                        "field": "title.trigram",
                        "suggest_mode": "always"
                    },
                    {
                        "field": "title.reverse",
                        "suggest_mode": "always",
                        "pre_filter": "reverse",
                        "post_filter": "reverse"
                    }
                ]
            }
        }
    }
}
```

I think the override is the more transparent approach. Am i missing something with the NameableInterface, do you have an example where that is used to build a list?

Reply:
@deguif Please run the tests whenever possible, i reverted to the `\Elastica\Suggest\Phrase::toArray` override.

Reply:
Yes done again, I did it this morning but Github was encountering issues.